src/Donations/Repositories/DonationRepository.php

```diff
@@ -468,11 +408,36 @@ private function validateDonation(Donation $donation)
             }
         }
 
+        if ( $donation->subscriptionId && $donation->type->isSingle()) {
+            throw new InvalidArgumentException('Subscription ID can only be set for recurring donations.');
+        }
+
+        if ( !$donation->subscriptionId && ( $donation->type->isRenewal() || $donation->type->isSubscription() ) ) {
+            throw new InvalidArgumentException('Subscription ID is required for recurring donations.');
+        }
+
```

Comment:
@JasonTheAdams I just ran into an issue submitting a recurring donation using stripe and this seems to be the culprit.  Remember, the donation is created _before_ the subscription right now so the `subscriptionId` is going to be null even if the type is subscription.

Reply:
In `LegacyPaymentGatewayAdapter->handleBeforeGateway` we're trying to save the status as subscription before the subscription exists so there is no subscriptionId

Reply:
Resolved in 07edc009f